core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1140,9 +1141,9 @@ private[kafka] class Processor(
                 expiredConnectionsKilledCount.record(null, 1, 0)
               } else {
                 val connectionId = receive.source
-                val context = new RequestContext(header, connectionId, channel.socketAddress,
-                  channel.principal, listenerName, securityProtocol,
-                  channel.channelMetadataRegistry.clientInformation, isPrivilegedListener, channel.principalSerde)
+                val context = new RequestContext(header, connectionId, channel.socketAddress, Optional.of(channel.socketPort()),
```

Comment:
Hmm, channel.socketPort() is never null. Why do we need to convert it to an Optional?

Reply:
I find RequestContext is either created in SocketServer or during forwarding requests. We currently don't require client port information outside KIP-714, push telemetry request is not marked forwardable, wiring the client port information elsewhere seems not useful at this point of time hence I marked it as Optional.